api/src/test/java/org/openmrs/api/db/UserDAOTest.java

```diff
@@ -178,4 +179,28 @@ public void isSecretAnswer_shouldReturnFalseWhenTheAnswerDoesNotMatch() {
 		
 	}
 	
+	@Test
+	public void getLoginCredentialByActivation_shouldGetLoginCredentialByActivationKey() {
+		String key="h4ph0fpNzQCIPSw8plJI";
+		int validTime = 10*60*1000; //equivalent to 10 minutes for token to be valid
+		Long tokenTime = System.currentTimeMillis() + validTime;
+		LoginCredential credentials = dao.getLoginCredential(userJoe);
+		credentials.setActivationKey("b071c88d6d877922e35af2e6a90dd57d37ac61143a03bb986c5f353566f3972a86ce9b2604c31a22dfa467922dcfd54fa7d18b0a7c7648d94ca3d97a88ea2fd0:"+tokenTime);			
+		dao.updateLoginCredential(credentials);	
+		LoginCredential loginCred = dao.getLoginCredentialByActivationKey(key);
+		assertNotNull("LoginCredential with activationKey ", loginCred); 	
+	}
+	
+	@Test
+	public void verifyUserActivationKey_shouldReturnNullIfActivationKeyIsNotValid() {
```

Comment:
This test doesn't make sense for the DAO, there is no logic in the DAO that needs to be tested since you are already testing the service layer methods anyways